src/test/java/net/sf/jabref/remote/DBPProcessorTest.java

```diff
@@ -0,0 +1,351 @@
+import java.util.Map;
+import java.util.Set;
+
+import net.sf.jabref.model.entry.BibEntry;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class DBPProcessorTest {
+
+    private static Connection connection;
+    private DBProcessor dbProcessor;
+    private DBHelper dbHelper;
+
+    @Parameter
+    public DBType dbType;
+
+
+    @Before
+    public void setUp() {
+
+        // Get only one connection for each parameter
+        if (TestConnector.currentConnectionType != dbType) {
+            try {
+                connection = TestConnector.getTestConnection(dbType);
+            } catch (Exception e) {
+                Assert.fail(e.getMessage());
+            }
+        }
+        dbProcessor = new DBProcessor(connection, dbType);
+        dbHelper = new DBHelper(connection);
+        dbProcessor.setUpRemoteDatabase();
+
+    }
+
+    @Parameters(name = "Test with {0} database system")
+    public static Collection<DBType> getTestingDatabaseSystems() {
+        Set<DBType> dbTypes = new HashSet<>();
+        dbTypes.add(DBType.MYSQL);
+        dbTypes.add(DBType.POSTGRESQL);
+
+        try {
+            Class.forName("oracle.jdbc.driver.OracleDriver");
+            dbTypes.add(DBType.ORACLE);
+        } catch (ClassNotFoundException e) {
+            // In case that Oracle interface is not available do not perform tests for this system.
+            System.out.println("Oracle driver not available. Skipping tests for this system...");
+        }
+        return dbTypes;
+    }
+
+    @Test
+    public void testCheckBaseIntegrity() {
+        Assert.assertTrue(dbProcessor.checkBaseIntegrity());
+        clear();
+        Assert.assertFalse(dbProcessor.checkBaseIntegrity());
+    }
+
+    @Test
+    public void testSetUpRemoteDatabase() {
+        clear();
+        dbProcessor.setUpRemoteDatabase();
+        Assert.assertTrue(dbProcessor.checkBaseIntegrity());
+    }
+
+    @Test
+    public void testInsertEntry() {
+
+        BibEntry realEntry = new BibEntry();
+        realEntry.setType("inproceedings");
+        realEntry.setField("author", "Wirthlin, Michael J and Hutchings, Brad L and Gilson, Kent L");
+        realEntry.setField("title", "The nano processor: a low resource reconfigurable processor");
+        realEntry.setField("booktitle", "FPGAs for Custom Computing Machines, 1994. Proceedings. IEEE Workshop on");
+        realEntry.setField("year", "1994");
+        realEntry.setCiteKey("nanoproc1994");
+        realEntry.setRemoteId(1);
+
+        dbProcessor.insertEntry(realEntry);
+
+        BibEntry emptyEntry = new BibEntry();
+        emptyEntry.setRemoteId(1);
+        dbProcessor.insertEntry(emptyEntry); // does not insert, due to same remoteId.
```

Comment:
If you look closer at the line containing `Assert.assertFalse(resultSet.next());`, you'll see that this makes shure, that no other rows/entries are present. I could split that up, but the second method would look like the first one.